pkg/config/config_template.yaml

```diff
@@ -1,858 +1,1197 @@
+## Uncomment this parameter and the one below to enable them.
+## See https://docs.datadoghq.com/agent/apm/
 #
-#   Enable container log collection for all the containers (see ac_exclude to filter out containers)
-#   container_collect_all: false
-#
-#   Define the endpoint and port to hit when using a proxy for logs. The logs are forwarded in TCP
-#   therefore the proxy must be able to handle TCP connections.
-#   logs_dd_url: <endpoint>:<port>
-#
-#   Disable the SSL encryption (default to false). This parameter should only be used when logs are
-#   forwarded locally to a proxy. It is highly recommended to then handle the SSL encryption
-#   on the proxy side.
-#   logs_no_ssl: false
-#
-#   Global processing rules that are applied to all the logs. The available rules are
-#   "exclude_at_match", "include_at_match" and "mask_sequences". More information in the documentation:
-#   https://docs.datadoghq.com/logs/log_collection/?tab=tailexistingfiles#advanced-log-collection-functions
-#   processing_rules:
-#     - rule1_arg1
-#       rule1_arg2
-#     - rule2_arg1
-#       rule2_arg2
-#
-#   By default, logs are sent to port 10516 (for the US site), use this parameter
-#   to force the agent to send logs in TCP to port 443 (default is false)
-#   use_port_443: false
+# apm_config:
+
+  ## @param enabled - boolean - optional - default: true
+  ## Set to true to enable the APM Agent.
+  #
+  # enabled: true
+
+  ## @param env - string - optional - default: none
+  ## The environment tag that Traces should be tagged with.
+  ## Inherits from "env" tag if "none" is applied here.
+  #
+  # env: none
+
+  ## @param receiver_port - integer - optional - default: 8126
+  ## The port that the trace receiver should listen on.
+  #
+  # receiver_port: 8126
+
+  ## @param apm_non_local_traffic - boolean - optional - default: false
+  ## Set to true so the Trace Agent listens for non local traffic,
+  ## i.e if Traces are being sent to this Agent from another host/container
+  #
+  # apm_non_local_traffic: false
+
+  ## @param extra_sample_rate - float - optional - default: 1.0
+  ## Extra global sample rate to apply on all the traces
+  ## This sample rate is combined to the sample rate from the sampler logic, still promoting interesting traces.
+  ## From 1 (no extra rate) to 0 (don't sample at all)
+  #
+  # extra_sample_rate: 1.0
+
+  ## @param max_traces_per_second - integer - optional - default: 10
+  ## Maximum number of traces per second to sample. The limit is applied over an average over
+  ## a few minutes ; much bigger spikes are possible. Set to 0 to disable the limit.
+  #
+  # max_traces_per_second: 10
+
+  ## @param ignore_resources - list of strings - optional
+  ## A blacklist of regular expressions can be provided to disable certain traces based on their resource name
+  ## all entries must be surrounded by double quotes and separated by commas.
+  #
+  # ignore_resources: ["(GET|POST) /healthcheck"]
```

Comment:
I think we're missing new configurations that are still important: https://github.com/DataDog/datadog-agent/blob/master/docs/trace-agent/datadog.example.yaml#L106-L150

* `max_memory` and `max_cpu_percent` are needed when you want an Agent to use all the host resources (i.e. you have dedicated hosts for the Trace Agent)
* `obfuscation` is an advanced configuration to fine-tune the obfuscation on our agent.
* `replace_tags` is to redact specific tags: https://github.com/DataDog/datadog-agent/blob/master/docs/trace-agent/datadog.example.yaml#L89-L98

I'd like to see those options here. Even if they're advanced, someone may be interested in using those.

Reply:
I thinks obfuscation may be too complex to be documented directly in the configuration file, I'm going to create a dedicated Guide for it that will live in the Public documentation